### PR TITLE
Recurrent doc fixes

### PR DESCRIPTION
**Pull request by skaae**
- Fix string representation of function in recurent layers
- add Gate to docs
- add Gate to recurrent docstring
- Change RecurrentLayer init to be normal(0.1) like all other layers
- add high forget gate bias example to gate class

### CODE REVIEW

File: lasagne/layers/recurrent.py [hunk 1]
```diff
@@ -379,8 +404,8 @@ class RecurrentLayer(CustomRecurrentLayer):
            arXiv preprint arXiv:1308.0850 (2013).
     """
     def __init__(self, incoming, num_units,
-                 W_in_to_hid=init.Uniform(),
-                 W_hid_to_hid=init.Uniform(),
+                 W_in_to_hid=init.Normal(0.1),
```
Comment (craffel) [inline lasagne/layers/recurrent.py]: Can we address this in a different issue/PR?  I think being consistent is good, but we are also not being consistent with the rest of `lasagne` if we do this.

File: lasagne/layers/recurrent.py [hunk 2]
```diff
@@ -438,6 +468,24 @@ class Gate(object):
     nonlinearity : callable or None
         The nonlinearity that is applied to the input gate activation. If None
         is provided, no nonlinearity will be applied.
+
+    References
```
Comment (craffel) [inline lasagne/layers/recurrent.py]: Let's use a different reference; this was actually proposed in the original forget gate paper: http://cognet.mit.edu/system/cogfiles/journalpdfs/089976600300015015.pdf . Can we also move the references under the example?

**Comment by craffel** [comment]
Thanks, didn't realize layers had started to have their signature repeated like this.

File: lasagne/layers/recurrent.py (hunk 2 above)
Comment (skaae) [inline lasagne/layers/recurrent.py]: It was the only reference I could remember. Can you write the name of the reference? My ip is blocked?

File: lasagne/layers/recurrent.py (hunk 1 above)
Comment (skaae) [inline lasagne/layers/recurrent.py]: yes.

File: lasagne/layers/recurrent.py (hunk 2 above)
Comment (craffel) [inline lasagne/layers/recurrent.py]: Sorry, my mistake for browsing refs while on Columbia's network.  The paper is "Learning to forget: continual prediction with LSTM" by Gers.  Using an initial forget gate bias of 5 is mentioned on page 15.

**Comment by skaae** [comment]
I changed the reference and moved it + use Uniform init again.

btw have you tried one of the conv test failing randomly? 

```
lasagne/tests/test_theano_extensions.py::test_conv[2-conv1d_sc] FAILED
```

It failed on my first run, but when i ran the tests again it passed?

File: lasagne/layers/recurrent.py [hunk 3]
```diff
@@ -321,7 +338,15 @@ def step_masked(input_n, mask_n, hid_previous, *args):
 
 
 class RecurrentLayer(CustomRecurrentLayer):
-    """Dense recurrent neural network (RNN) layer
+    """
+    lasagne.layers.recurrent.RecurrentLayer(incoming, num_units,
+    W_in_to_hid=init.Uniform(), W_hid_to_hid=init.Uniform(),
```
Comment (craffel) [inline lasagne/layers/recurrent.py]: Shouldn't these be `lasagne.init.Uniform()`?

**Comment by craffel** [comment]
I just made a small docstring update and rebased against upstream/master, so you'll need to do a rebase.  Sorry!

File: lasagne/layers/recurrent.py (hunk 3 above)
Comment (skaae) [inline lasagne/layers/recurrent.py]: Yes. I'll rebase. 

Shouldnt i squash everything into a single commit?

**Comment by skaae** [comment]
rebased, squashed and fixed the error you spotted.

**Comment by skaae** [comment]
I was a little too enthusiastic when cleaning up branches..

**Comment by craffel** [comment]
Hm, it's a little weird that when you compile the docs the "References" are forced to show up above the "Examples" section, but I suppose there's not a lot we can do about that.  Merged in d695a394b22e854e40f37d156b41b5696cbdf375, thanks!

**Pull request closed without merging**